src/postamble.js

```diff
@@ -225,7 +225,7 @@ function run(args) {
   if (preloadStartTime === null) preloadStartTime = Date.now();
 
   if (runDependencies > 0) {
-#if ASSERTIONS
+#if ASSERTIONS && !BINARYEN_ASYNC_COMPILATION
```

Comment:
Not quite, since startup can be async for other reasons as well, in particular the mem init file for asm.js is loaded separately. That's an async operation that happens by default in -O2+. Also, if files are preloaded, that will make startup async as well.

Instead of trying to cover all those cases, I think it's simpler to just put logging like this behind another flag (as even in async startup, it's useful for debugging).